omniscidb/QueryEngine/Compiler/genx.ll

```diff
@@ -50,3 +49,29 @@ define i64 @agg_count(i64* %agg, i64 %val) {
     %old = atomicrmw add i64* %agg, i64 1 monotonic
     ret i64 %old
 }
+
+define i64 @agg_count_skip_val(i64* %agg, i64 noundef %val, i64 noundef %skip_val) {
+    %no_skip = icmp ne i64 %val, %skip_val
+    br i1 %no_skip, label %.noskip, label %.skip
+.noskip:
+;;  todo: %old = call i64 @agg_count(i64* %agg, i64 %val)
+    %ld = load i64, i64* %agg
+    %old = atomicrmw add i64* %agg, i64 1 monotonic
```

Comment:
do we need this atomic?

Reply:
Not really, a regular `add` works just fine here. At least for non-grouped aggregations that do not use shared memory ;). I plan to revisit it once we have shared memory turned on (which I'm working on now).